contributors/design-proposals/apiserver-build-in-admission-plugins.md

```diff
@@ -0,0 +1,58 @@
+# Build some Admission Controllers into the Generic API server library
+
+**Related PR:**
+
+| Topic | Link |
+| ----- | ---- |
+| Admission Control | https://github.com/kubernetes/community/blob/master/contributors/design-proposals/admission_control.md |
+
+## Introduction
+
+An admission control is a piece of code that intercepts requests to the Kubernetes API - think a middleware.
+The API server lets you have a whole chain of them. Each is run in sequence before a request is accepted 
+into the cluster. If any of the plugins in the sequence reject the request, the entire request is rejected 
+immediately and an error is returned to the user.
+
+Many features in Kubernetes require an admission control plugin to be enabled in order to properly support the feature. 
+In fact in the [documentation](https://kubernetes.io/docs/admin/admission-controllers/#is-there-a-recommended-set-of-plug-ins-to-use) you will find 
+a recommended set of them to use.
+
+At the moment admission controllers are implemented as plugins and they have to be compiled into the 
+final binary in order to be used at a later time. Some even require an access to cache, an authorizer etc.
+This is where an admission plugin initializer kicks in. An admission plugin initializer is used to pass additional 
+configuration and runtime references to a cache, a client and an authorizer.
+
+To streamline the process of adding new plugins especially for aggregated API servers we would like to build some plugins 
+into the generic API server library and provide a plugin initializer. While anyone can author and register one, having a known set of 
+provided references let's people focus on what they need their admission plugin to do instead of paying attention to wiring.
+
+## Implementation
+
+The first step would involve creating a "standard" plugin initializer that would be part of the 
```

Comment:
It will include a few shared resources, not an associative array.  For anything beyond the normal set, they'll have to add their own PluginInitializer to the chain.